### ISSUE TITLE
`NAME` is recognized instead of `NAME.github.io`

### ISSUE THREAD

**Issue by fu-sen** [desktop/desktop#11448]
### Describe the bug

If you personally have the repositories `NAME.github.io` and `NAME` , adding `NAME.github.io` will recognize `NAME` instead.

### Version & OS

- GitHub Desktop Version 2.6.2
- Windows 10 Home Version 20H2

### Steps to reproduce the behavior

You can reproduce this using Organization instead of a personal account

1. generates repositories `NAME.github.io`
2. Rename the repository of `NAME.github.io` to` NAME` (e.g. Repository page - Settings)
1. generates repositories `NAME.github.io` again
2. Select **Add**  and add `NAME.github.io` on GitHub Desktop.
3. Make sure the listed repositories are `NAME` instead of `NAME.github.io`

### Screenshots

Until recently I was successfully using `fu-sen.github.io` on GitHub Desktop. It was encountered with the addition of `fu-sen`.

![スクリーンショット 2021-01-24 11 16 15](https://user-images.githubusercontent.com/5434159/105619593-873bfd80-5e37-11eb-8019-3d6acf0d741a.jpg)
![スクリーンショット 2021-01-24 11 16 19](https://user-images.githubusercontent.com/5434159/105619597-8c00b180-5e37-11eb-8a46-17dc15f21419.jpg)

But `fu-sen.github.io/.git/config` successfully references` fu-sen.github.io`
(Refer to the folder (directory) in the title bar):
![スクリーンショット 2021-01-24 11 31 04](https://user-images.githubusercontent.com/5434159/105619631-f44f9300-5e37-11eb-948c-2a112611b975.jpg)

I added 2 repositories to Sourcetree and it's fine:
![スクリーンショット 2021-01-24 11 16 01](https://user-images.githubusercontent.com/5434159/105619671-5c9e7480-5e38-11eb-8275-ac8c73d03964.jpg)

Note: I'm Japanese, so I see a `￥` instead of a` ＼ `

### Actual behavior

Even if you encounter this problem, local Push etc. can be reflected in the repository `USER.github.io`.

But if you add the repository from GitHub Desktop, you will get the `USER` repository instead of` USER.github.io`.
This can be avoided by other means, such as `git clone` from the CLI.

### Logs

It contains an automatic update error, but paste it as is.

```
2021-01-23T02:36:14.361Z - info: [ui] [AppStore] loading 63 repositories from store
2021-01-23T02:36:14.362Z - info: [ui] [AppStore] found account: fu-sen (BALLOON | FU-SEN)
2021-01-23T02:36:14.693Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2021-01-23T02:36:14.794Z - info: [ui] launching: 2.6.2 (Windows 10.0.19042)
2021-01-23T02:36:14.795Z - info: [ui] execPath: 'C:\Users\balloon\scoop\apps\github\current\GitHubDesktop.exe'
2021-01-23T02:36:14.907Z - error: [ui] Error checking for updates
Error: The application is missing a dependency it needs to check and install updates. This is very, very bad.
    at file:///C:/Users/balloon/scoop/apps/github/2.6.2/resources/app/renderer.js:45:170322
    at onAutoUpdaterError (file:///C:/Users/balloon/scoop/apps/github/2.6.2/resources/app/renderer.js:45:170738)
    at CallbacksRegistry.apply (electron/js2c/renderer_init.js:131:792)
    at electron/js2c/renderer_init.js:83:5068
    at EventEmitter.<anonymous> (electron/js2c/renderer_init.js:83:4308)
    at EventEmitter.emit (events.js:223:5)
    at Object.onMessage (electron/js2c/renderer_init.js:115:818)
2021-01-23T02:36:20.947Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.615s)
2021-01-23T02:36:30.523Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2021-01-23T02:36:35.112Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2021-01-23T02:36:41.863Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 2.006s)
2021-01-23T02:36:43.538Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.191s)
2021-01-23T02:36:45.387Z - info: [ui] Executing getStatus: git --no-optional-locks status --untracked-files=all --branch --porcelain=2 -z (took 1.335s)
2021-01-23T02:37:02.692Z - info: [ui] [Timing] Action 'create commit' for 'fu-sen/fu-sen' took 2.110s
2021-01-23T02:37:02.714Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: fu-sen - fu-sen (has token)
2021-01-23T02:37:08.361Z - info: [ui] Executing push: git -c credential.helper= -c protocol.version=2 push origin main:main --progress (took 5.494s)
2021-01-23T02:37:10.185Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.681s)
2021-01-23T02:37:18.389Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 62 repositories took 45.5s of which 0.0s paused, total 45.5s
2021-01-23T02:37:50.373Z - info: [main] opening in browser: https://github.com/fu-sen/fu-sen
2021-01-23T02:38:31.372Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.611s)
2021-01-23T02:38:43.829Z - info: [ui] [AppStore] adding repository at C:\Users\balloon\Dropbox\GitHub-balloon.vg\fu-sen.github.io to store
2021-01-23T02:38:44.816Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2021-01-23T02:38:50.635Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
2021-01-23T02:38:52.542Z - info: [ui] Executing fetch: git -c credential.helper= -c protocol.version=2 fetch --progress --prune origin (took 1.745s)
2021-01-23T02:38:53.628Z - info: [ui] [BranchPruner] Last prune took place in a day - skipping
```

### Additional context

For personal accounts, `NAME` has become a special repository.
Therefore, when you rename from Personal GitHub Page and generate a special repository, you are more likely to encounter the bug.

The same issue was reported before, but it seems I couldn't reproduce it:
https://github.com/desktop/desktop/issues/8425

This may be encountered only for some users or OS (Windows).

**Comment by sergiou87** [comment]
Thank you for your report @fu-sen !!

I tried on Windows and macOS but I haven't been able to reproduce this issue. Could you also show us how you add the repo to GitHub Desktop? Did you have the `fu-sen/fu-sen` repo already in the app?

A video of the whole process would be very helpful too.

**Comment by fu-sen** [comment]
@sergiou87 Thank you for your confirmation. 😃 It was the same result as https://github.com/desktop/desktop/issues/8425  
It seems to be happening to a particular user, but it seems difficult to pinpoint the condition. 🤔
I haven't shot the video on my PC, so I need some time to prepare it. Please wait. ☕

**Comment by fu-sen** [comment]
I recorded a video:
https://user-images.githubusercontent.com/5434159/105694812-b4c29d00-5f44-11eb-9293-7e45e78ce0ae.mp4

**Comment by sergiou87** [comment]
That is super useful, thank you!! I'll see if I can find where the problem is exactly 😄

**Comment by fu-sen** [comment]
This bug seems to be difficult to catch... Good luck.

**Comment by sergiou87** [comment]
Yeah I can't reproduce the problem here, super weird O_o

@fu-sen can you try something for me?
1. Open the Developer Tools: `View` -> `Toggle Developer Tools`
2. Switch to the `Sources` tab in the developer tools.
3. Press `Control + P` and type `cloning-repositories-store.ts` to open that file in the developer tools.
4. Click on the margin of line 26 to set a breakpoint there.
5. Then, try to clone your `fu-sen.github.io` repo again to reproduce the issue.
6. When the app pauses on the breakpoint you set, you can either press Esc or switch to the `Console` tab, and type first `path` and press enter, then type `url` and press enter, and then `repository.name` and press enter.

You should get something like this:

![image](https://user-images.githubusercontent.com/1083228/105709560-2afbdb00-5f16-11eb-816e-aafb0fe01935.png)

I wonder what those 3 values look like for you 🤔 
After you're done you can click again in the red dot 🔴  to remove the breakpoint.

**Comment by fu-sen** [comment]
OK. I did. It was easy to operate because I've seen it in Chrome 😉 :

**type `path`** :
```
execPath: 'C:\Users\balloon\scoop\apps\github\current\GitHubDesktop.exe'
```

**type `url`** :
```
[AppStore.getAccountForRemoteURL] account found for remote: https://github.com/fu-sen/fu-sen.git - fu-sen (has token)
```

**type `repository.name`** :
No results

**Console full log, because it's short** :
```
install.ts:31 [AppStore] loading 64 repositories from store
install.ts:31 [AppStore] found account: fu-sen (BALLOON | FU-SEN)
install.ts:31 [BranchPruner] Last prune took place in 21 hours - skipping
app.tsx:252 App ready time: 76.98519998788834ms
app.tsx:253 Load time: 788.8703999966383ms
app.tsx:254 Renderer ready time: 716.7800000170246ms
install.ts:31 launching: 2.6.2 (Windows 10.0.19042)
install.ts:31 execPath: 'C:\Users\balloon\scoop\apps\github\current\GitHubDesktop.exe'
install.ts:23 Error checking for updates
Error: The application is missing a dependency it needs to check and install updates. This is very, very bad.
    at squirrel-error-parser.ts:19
    at onAutoUpdaterError (update-store.ts:87)
    at CallbacksRegistry.apply (electron/js2c/renderer_init.js:131)
    at electron/js2c/renderer_init.js:83
    at EventEmitter.<anonymous> (electron/js2c/renderer_init.js:83)
    at EventEmitter.emit (events.js:223)
    at Object.onMessage (electron/js2c/renderer_init.js:115)
error @ renderer.js:45
install.ts:31 [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/fu-sen/fu-sen.git - fu-sen (has token)
```

This operation cloned the `fu-sen` repository. That is the worst result we expected:
![スクリーンショット 2021-01-25 22 40 10](https://user-images.githubusercontent.com/5434159/105713638-bac57780-5f5e-11eb-9b16-d75e398e5d85.jpg)

**Comment by fu-sen** [comment]
Simply add the locally correct repository `fu-sen.github.io` with **Add existing Repository**
and GitHub Desktop will display` fu-sen`. It's a strange work. 😫

**Comment by sergiou87** [comment]
I can't reproduce the problem with `Add existing Repository` either 😢 

Regarding the values you posted up there, I think those are not what I expected to see: I think typing those values in the Console doesn't work for the production/beta versions of the app, sorry about that 😞  However, seems like hovering the mouse over `path`, `url` and `repository.name` a few lines below should work:

![image](https://user-images.githubusercontent.com/1083228/105722600-1f181500-5f26-11eb-8549-c44e84c4ab1d.png)
![image](https://user-images.githubusercontent.com/1083228/105722609-250df600-5f26-11eb-8463-65082315f59f.png)
![image](https://user-images.githubusercontent.com/1083228/105722660-30f9b800-5f26-11eb-83a0-70825c196fbc.png)

Does that work for you?

**Comment by fu-sen** [comment]
I realize there is a possibility and experiment with it. ✋

The repository `fu-sen` was generated from` fu-sen.github.io` by renaming the repository.
I know that when GitHub Desktop renames a repository, it keeps track of it and renames it to the new name.
I predict that the function is bad. So if `fu-sen.github.io` is now` fu-sen`, GitHub Desktop is erroneously acting from the history.

So, I will check if it can be improved by deleting the management of repository `fu-sen` once and regenerating it. There is no problem that this history disappears.

**Comment by fu-sen** [comment]
Alright! After deleting the repository `fu-sen` which was` fu-sen.github.io` in the past, the repository `fu-sen.github.io` was successfully cloned. 👍
![スクリーンショット 2021-01-26 00 21 06](https://user-images.githubusercontent.com/5434159/105725858-c5870900-5f6c-11eb-8b16-267ea270a4a4.jpg)

After that, a new repository `fu-sen` was generated. The two repositories are now healthy.
![スクリーンショット 2021-01-26 00 27 35](https://user-images.githubusercontent.com/5434159/105726515-81483880-5f6d-11eb-8b9a-16f19d046e2a.jpg)

This solved the problem myself, 😄 but it allows me to redefine an obvious bug in GitHub Desktop. Should I continue with this issue open? Or do we build a new issue?

**Comment by sergiou87** [comment]
That's very interesting! I'm very happy you could at least work around it 🎉 

Thanks for renaming the issue, if you could also update the original post to match your latest findings, that'd be awesome ❤️

**Comment by fu-sen** [comment]
Japan was at night at that time, so I slept. 🛌💤

@sergiou87 You can do other work until I update the information.

**Comment by fu-sen** [comment]
I did the following experiment:

1. Create repository `old`
2. Renamed `old` to` new`
3. Create repository `old` again
4. Add repository `old` to GitHub Desktop

It envisioned a bug encounter, but unfortunately GitHub Desktop cloned successfully.
So the conditions for encountering this bug need more information, but I can't add anything to it.

Many people haven't generated a Spacial repository, so other people may come across this hug in the future, so I can undo the title and close this issue once to get this information.

@sergiou87 Thank you for your support so far. 😄

**Comment by sergiou87** [comment]
Thanks for checking that too, @fu-sen !! I tried those same steps and, as you also saw, it worked perfectly fine. We'll dig more when someone runs into it again.

**Comment by fu-sen** [comment]
I'm ready to reopen. 💻 This could be reproduced in the Organization, so you can verify it in the Organization.

I edited the original issue text, but I'll quote the steps here as well:

> ## Steps to reproduce the behavior
>
> You can reproduce this using Organization instead of a personal account
>
> 1. generates repositories `NAME.github.io`
> 2. Rename the repository of `NAME.github.io` to `NAME` (e.g. Repository page - Settings)
> 3. generates repositories `NAME.github.io` again
> 4. Select Add and add `NAME.github.io` on GitHub Desktop.
> 5. Make sure the listed repositories are `NAME` instead of `NAME.github.io`

I hope you will encounter a bug with this.

**Comment by fu-sen** [comment]
@sergiou87 If I reopen this, are you responsible for this issue again? This seems to be overlooked.

**Comment by sergiou87** [comment]
I'm sorry @fu-sen, I moved my focus to other issues. Someone in the team (which could still be me 😆 ) will take a look when we have a chance.

Thanks for updating with your latest findings! ❤️

**Comment by sergiou87** [comment]
I think this should've been fixed by #12308 😄
